src/components/Ffuf/FfufTool.tsx

```diff
@@ -22,38 +25,47 @@ const description_userguide =
     "       E.g. wordlist.txt\n\n" +
     "Step 3: Optionally enter extensions to be added to words. Comma separated." +
     "\n         E.g. .html,.php,.txt" +
-    "\n\nStep 4: Click Scan to commence the ffuf operation.\n\n" +
+    "\n\nStep 4: Click Scan to commence the ffuf operation.\n" +
     "Step 5: View the Output block below to view the results of the Scan.";
 
+/**
+ * Represents the form values for the Ffuf component.
+ */
 interface FormValuesType {
-    wordlist: string;
+    wordList: string;
     url: string;
     output: string;
-    filtersize: string;
+    filterSize: string;
     extensions: string;
-    filterwords: string;
-    filterlines: string;
+    filterWords: string;
+    filterLines: string;
 }
 
+/**
+ * The Ffuftool component.
+ * @returns The Ffuftool component.
+ */
 const FfufTool = () => {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
+    // Component State Variables
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
     const [checkedStopWhen, setCheckedStopWhen] = useState(false);
     const [checkedVerboseOutput, setCheckedVerboseOutput] = useState(false);
     const [checkedAdvanced, setCheckedAdvanced] = useState(false);
-    const [pid, setPid] = useState("");
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
 
+    // Form Hook to handle form input.
     let form = useForm({
         initialValues: {
-            wordlist: "",
+            wordList: "",
             url: "",
             output: "",
-            filtersize: "",
+            filterSize: "",
             extensions: "",
-            filterwords: "",
-            filterlines: "",
+            filterWords: "",
+            filterLines: "",
         },
     });
 
```

Comment:
This is where useEffect and checkAllCommandAvailability should go. Refer to Airbase-NG for an example.